src/main/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/StringToCodeWithAuthorityTypeConverter.java

```
@@ -0,0 +1,37 @@
+package au.gov.ga.geodesy.support.mapper.orika.geodesyml;
+
+import ma.glasnost.orika.MappingContext;
+import ma.glasnost.orika.converter.BidirectionalConverter;
+import ma.glasnost.orika.metadata.Type;
+import net.opengis.gml.v_3_2_1.CodeType;
+import net.opengis.gml.v_3_2_1.CodeWithAuthorityType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StringToCodeWithAuthorityTypeConverter<T extends CodeWithAuthorityType> extends BidirectionalConverter<String, T> {
+    Logger logger = LoggerFactory.getLogger(StringToCodeWithAuthorityTypeConverter.class);
+    private String codeSpace;
+
+    public StringToCodeWithAuthorityTypeConverter(String codeSpace) {
+        this.codeSpace = codeSpace;
+    }
+
+    public String convertFrom(T code, Type<String> targetType, MappingContext ctx) {
+        return code.getValue();
+    }
+
+    public T convertTo(String str, Type<T> targetType, MappingContext ctx) {
+        T code = null;
+        try {
+            code = targetType.getRawType().newInstance();
+        } catch (InstantiationException | IllegalAccessException e) {
```

I think that landing in this catch block is an indication of a bug that we need to know about. It is better to throw a subclass of runtime exception, rather than recover by returning null.